plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/TransactionLogTail.java

```diff
@@ -146,24 +146,13 @@ public static Optional<List<DeltaLakeTransactionLogEntry>> getEntriesFromJson(lo
             return Optional.of(resultsBuilder.build());
         }
         catch (IOException e) {
-            if (isFileNotFoundException(e)) {
+            if (e instanceof FileNotFoundException) {
```

Comment:
Also here the `FileNotFoundException` handling could be extracted to a different `catch`.